driver-core/src/main/java/com/datastax/driver/core/querybuilder/BuiltStatement.java

```diff
@@ -297,14 +296,16 @@ public Boolean isIdempotent() {
     @Override
     public String toString() {
         try {
-            if (forceNoValues) {
-                return getQueryString();
-            }
-            StringBuilder queryString = buildQueryString(null, CodecRegistry.DEFAULT_INSTANCE);
-            return maybeAddSemicolon(queryString).toString();
+            // try first with all values inlined (will not work if some values require custom codecs)
```

Comment:
What are the implications of no longer check forceNoValues?  Isn't that here for protocol version 1?  Although toString (maybe) isn't used by the driver itself, so it is not considered relevant here?